pkg/service/endpoints_controller.go

```diff
@@ -127,24 +126,24 @@ func (e *EndpointController) SyncServiceEndpoints() error {
 	return resultErr
 }
 
-func containsEndpoint(endpoints *api.Endpoints, endpoint string) bool {
-	if endpoints == nil {
+func containsEndpoint(haystack *api.Endpoints, needle *api.Endpoint) bool {
```

Comment:
I generally agree with that sentiment, except that "haystack" and "needle" go back to K&R C's definition of strstr() - in this case I think it improves readability.  I could be haggled down to something like allEndpoints and endpoint, if you feel stringly.